test/widgets/app_test.dart

```diff
@@ -51,4 +54,113 @@ void main() {
       ]);
     });
   });
+
+  group('ChooseAccountPage', () {
+    Future<void> setupChooseAccountPage(WidgetTester tester, {
+      required List<Account>? accounts,
+    }) async {
+      addTearDown(testBinding.reset);
+
+      for (final account in accounts ?? []) {
+        await testBinding.globalStore
+          .insertAccount(account.toCompanion(false));
+      }
+
+      await tester.pumpWidget(
+        const MaterialApp(
+          localizationsDelegates: ZulipLocalizations.localizationsDelegates,
+          supportedLocales: ZulipLocalizations.supportedLocales,
+          home: GlobalStoreWidget(
+            child: ChooseAccountPage())));
+
+      // global store gets loaded
+      await tester.pumpAndSettle();
+    }
+
+    List<Account> generateAccounts(int count) {
+      return List.generate(count, (i) => eg.account(
+        id: i,
+        user: eg.user(fullName: 'User $i', email: 'user$i@example'),
+        apiKey: 'user${i}apikey',
+      ));
+    }
+
+    Finder findAccount(Account account) => find.text(account.email).hitTestable();
+
+    Finder findButton<T extends ButtonStyleButton>({required String withText}) {
+      return find
+        .descendant(of: find.bySubtype<T>(), matching: find.text(withText))
+        .hitTestable();
+    }
+
+    void checkAccountShown(Account account, {required bool expected}) {
+      check(findAccount(account).evaluate()).length.equals(expected ? 1 : 0);
+    }
+
+    void checkButtonShown<T extends ButtonStyleButton>({
+      required String withText,
+      required bool expected,
+    }) {
+      check(findButton<T>(withText: withText).evaluate())
+        .length.equals(expected ? 1 : 0);
+    }
+
+    testWidgets('accounts list is scrollable when more than a screenful',
+      (tester) async {
```

Comment:
nit: we just leave the `(tester) async {` part on the same line even if that makes the line long:

```suggestion
    testWidgets('accounts list is scrollable when more than a screenful', (tester) async {
```

Because that part is totally boring and predictable, it's OK if it gets scrolled off screen for the reader.